call-home/src/bin/callhome/main.rs

```diff
@@ -256,5 +294,84 @@ async fn generate_report(
     report.storage_media = StorageMedia::new(valid_disks);
 
     report.nexus = Nexus::new(event_data);
+
+    let logs = logs.lock().unwrap();
+    // Taking a slice of the last hundred logs messages, to not make the payload too big.
```

Comment:
Shouldn't this be handled in the log side? Otherwise this will keep growing forever?

Reply:
I've added the length check while preparing the logs buffer. I'm doing it using a new function `rev_length_check`. I've removed the slicing when pushing to the Report.